packages/combo-box/src/vaadin-combo-box-data-provider-mixin.js

```diff
@@ -210,32 +173,63 @@ export const ComboBoxDataProviderMixin = (superClass) =>
         return;
       }
 
-      this._pendingRequests = {};
-      const filteredItems = [];
-      for (let i = 0; i < (this.size || 0); i++) {
-        filteredItems.push(this.__placeHolder);
-      }
-      this.filteredItems = filteredItems;
+      this.__dataProviderController.clearCache();
+
+      this.requestContentUpdate();
 
       if (this._shouldFetchData()) {
         this._forceNextRequest = false;
-        this._loadPage(0);
+        this.__dataProviderController.loadFirstPage();
       } else {
         this._forceNextRequest = true;
       }
     }
 
     /** @private */
-    _sizeChanged(size = 0) {
-      const filteredItems = (this.filteredItems || []).slice(0, size);
-      for (let i = 0; i < size; i++) {
-        filteredItems[i] = filteredItems[i] !== undefined ? filteredItems[i] : this.__placeHolder;
+    _sizeChanged(size) {
+      // When the size update originates externally, sync the new size with
+      // the controller and request a content update to re-render the scroller.
+      const { rootCache } = this.__dataProviderController;
+      if (rootCache.size !== size) {
+        rootCache.size = size;
+        // The controller adds new placeholders to the cache through mutation,
+        // so we need to create a new array to trigger filteredItems observers.
+        rootCache.items = [...rootCache.items];
```

Comment:
Seems like this line is untested (all the tests still pass when I remove it).

Reply:
Yeah, right, this piece might not actually be necessary, but it’s here to ensure backward compatibility with the previous version of this method, which created a new array whenever there was a size change. There is a slim chance some workarounds could rely on this behavior. So, I'd keep this change just to be on the safe side even though it's not covered. We could probably remove it later, with a warning about a potential breaking change, when we improve requestContentUpdate to properly handle mutations.

Reply:
Makes sense. Let's keep this line then.